---
title: "Qiqi Xie's CV"
author: Qiqi Xie
date: "`r Sys.Date()`"
output:
  pagedown::html_resume:
    css: ['styles.css']
    # set it to true for a self-contained HTML page but it'll take longer to render
    self_contained: true
---

```{r, include=FALSE}
knitr::opts_chunk$set(
  results='asis', 
  echo = FALSE
)


CRANpkg <- function (pkg) {
    cran <- "https://CRAN.R-project.org/package"
    fmt <- "[%s](%s=%s)"
    sprintf(fmt, pkg, cran, pkg)
}

Biocpkg <- function (pkg) {
    sprintf("[%s](http://bioconductor.org/packages/%s)", pkg, pkg)
}

library(glue)
library(tidyverse)

# Set this to true to have links turned into footnotes at the end of the document
PDF_EXPORT <- FALSE

# Holds all the links that were inserted for placement at the end
links <- c()

find_link <- regex("
  \\[   # Grab opening square bracket
  .+?   # Find smallest internal text as possible
  \\]   # Closing square bracket
  \\(   # Opening parenthesis
  .+?   # Link text, again as small as possible
  \\)   # Closing parenthesis
  ",
  comments = TRUE)

sanitize_links <- function(text){
  if(PDF_EXPORT){
    str_extract_all(text, find_link) %>% 
      pluck(1) %>% 
      walk(function(link_from_text){
        title <- link_from_text %>% str_extract('\\[.+\\]') %>% str_remove_all('\\[|\\]') 
        link <- link_from_text %>% str_extract('\\(.+\\)') %>% str_remove_all('\\(|\\)')
        
        # add link to links array
        links <<- c(links, link)
        
        # Build replacement text
        new_text <- glue('{title}<sup>{length(links)}</sup>')
        
        # Replace text
        text <<- text %>% str_replace(fixed(link_from_text), new_text)
      })
  }
  
  text
}


# Takes a single row of dataframe corresponding to a position
# turns it into markdown, and prints the result to console.
build_position_from_df <- function(pos_df){
  
  missing_start <- pos_df$start == 'N/A'
  dates_same <- pos_df$end == pos_df$start
  if (pos_df$end == 9999) {
    pos_df$end = "present"
  }
  if(any(c(missing_start,dates_same))){
    timeline <- pos_df$end
  } else {
    timeline <- glue('{pos_df$end} - {pos_df$start}')
  }

  descriptions <- pos_df[str_detect(names(pos_df), 'description')] %>% 
    as.list() %>% 
    map_chr(sanitize_links)
  
  # Make sure we only keep filled in descriptions
  description_bullets <- paste('-', descriptions[descriptions != 'N/A'], collapse = '\n')
  
  if (length(description_bullets) == 1 && description_bullets == "- ") {
    description_bullets <- ""
  }
  glue(
"### {sanitize_links(pos_df$title)}

{pos_df$loc}

{pos_df$institution}

{timeline}

{description_bullets}


"
  ) %>% print()
}

# Takes nested position data and a given section id 
# and prints all the positions in that section to console
print_section <- function(position_data, section_id){
  x <- position_data %>% 
    filter(section == section_id) %>% 
    pull(data) 
  
  prese <- " - "
  xx <- list()

  for (i in seq_along(x)) {    
      y = x[[i]]
      y <- cbind(y, start2 = as.character(y$start))
      y <- cbind(y, end2 = as.character(y$end))

      se <- paste(y$start, "-", y$end, collapse = " ")
      if (prese == se) {
        y$start2 = ""
        y$end2 = ""
      } else {
        prese = se
      }

    xx[[i]] <- select(y, -c(start, end)) %>%
      rename(start=start2, end=end2)
  }
    
  xx %>% 
    purrr::walk(build_position_from_df)
}



fill_nas <- function(column){
  ifelse(is.na(column), 'N/A', column)
}

# Load csv with position info
position_data <- read_csv('positions.csv') %>% 
  mutate_all(fill_nas) %>% 
  arrange(order, desc(end)) %>% 
  mutate(id = 1:n()) %>% 
  nest(data = c(-id, -section))
```

```{r}
# When in export mode the little dots are unaligned, so fix that. 
if(PDF_EXPORT){
  cat("
  <style>
  :root{
    --decorator-outer-offset-left: -6.5px;
  }
  </style>")
}
```


Aside
================================================================================


![logo](QIQIXIE.jpg){width=100%}



```{r}
# When in export mode the little dots are unaligned, so fix that. 
if(PDF_EXPORT){
  cat("View this CV online with links at _guangchuangyu.github.io/cv_")
}
```

Contact {#contact}
--------------------------------------------------------------------------------


- <i class="fa fa-envelope"></i> QiqiXie@iu.edu
- <i class="fa fa-twitter"></i> QiqiXie
- <i class="fa fa-github"></i> https://github.com/Qiqi-Xie
- <i class="fa fa-link"></i> https://orcid.org/0000-0003-4099-5287 
- <i class="fa fa-weixin"></i> lcy505264
- <i class="fa fa-phone"></i> (86) 15719612948
- <i class="fa fa-jianshu"></i> https://www.jianshu.com/u/bcb81276c29d
- <i class="fa fa-wolai"></i> https://www.wolai.com/qiqixie


![logo](ORCID_QIQIXIE.png){width=20%}


<br>

```{r}
profile = jsonlite::fromJSON("profile.json")
cites = jsonlite::fromJSON("citation.json")
if (profile$total_cites < sum(cites$cites))
  profile$total_cites <- sum(cites$cites)

hcr_symbol = '<font color="#cc5500"><i class="fa fa-trophy"></i></font>'

glue(

"
+ {hcr_symbol} Researcher
+ Citation = {profile$total_cites}
+ H-index = {profile$h_index}
+ I10-index = {profile$i10_index}

"
  ) %>% print()

```

![](citation.png)

Main
================================================================================

Qiqi Xie {#title}
--------------------------------------------------------------------------------


```{r, results='asis'}
intro_text <- glue("Bioinformatics Scientist. Data scientist. Biologist. Master of Medicine.")

cat(sanitize_links(intro_text))
```

I received the B.S. degree in Bachelor of Medicine from Nanchang University, City, China, in 2015 and the M.S. degree in Clinical Surgery from Lanzhou University, City, China, in 2019. I am currently undertaking doctoral training at Indiana University Bloomington, specializing in cancer stem cells, and I plan to pursue postdoctoral training in the future. 

I am broadly interested in gbm, breast cancer, bone tumor, and neuropathic pain. Fluent in Bioinformatics, NGS analysis, scRNA-seq, and machine learning.


Education {data-icon=graduation-cap data-concise=true}
--------------------------------------------------------------------------------

```{r, results='asis', echo = FALSE}
print_section(position_data, 'education')
```


Certificate {data-icon=book}
--------------------------------------------------------------------------------

```{r, results='asis', echo = FALSE}
print_section(position_data, 'certificate')
```


Grants  {data-icon=chart-line}
--------------------------------------------------------------------------------

```{r}
print_section(position_data, 'grant')
```


Publications {data-icon=book}
--------------------------------------------------------------------------------


```{r}
print_section(position_data, 'academic_articles')
```



Conference proceedings {data-icon=group}
--------------------------------------------------------------------------------



```{r}
print_section(position_data, 'presentation')
```


```{r}
if(PDF_EXPORT){
  cat("
  
Links {data-icon=link}
--------------------------------------------------------------------------------



")
  
  walk2(links, 1:length(links), function(link, index){
    print(glue('{index}. {link}'))
  })
}
```


Professional skills {data-icon=book}
--------------------------------------------------------------------------------

**Wet-lab Experimental**

- I am good at Immunohistochemistry, Immunofluorescence, Western Blot, RT-PCR, RNAi, Exosome identification, Open-field Test, Water Maze Test, Forced Swimming Test, Von Frey Hair Test, Cold Hot Plate Test, and Tereotactic Injection.

**Experimental instruments**

- I can operate the following experimental instruments: micro-drop PCR system, automatic protein expression quantitative system, high-content live cell imaging quantitative analysis system and cell metabolic analyzers.

**Software**

- I have advanced experience in using Microsoft Office, Illustrator, Photoshop, GraphPad Prism, SPSS, ImageJ, Image-Pro Plus, Cytoscape, Funrich.

**Statistics**

- I have moderate experience in using R and Python for statistical modeling and data visualization.

**Machine learning**

- I know how to do machine learning (including deep learning) and have applied some technologies to my projects.

**Data analysis**

- I have advanced experience in using R and Shell for data preprocessing, data cleaning and data interpretation. RNA-seq, ChIP-seq, ATAC-seq, scRNA-seq, scATAC-seq Analysis. Spatial Transcriptomics.


Disclaimer {#disclaimer}
--------------------------------------------------------------------------------

Last updated on `r Sys.Date()`.
